src/main/java/games/resistance/ResGameState.java

```diff
@@ -0,0 +1,283 @@
+package games.resistance;
+
+import core.AbstractGameState;
+import core.AbstractParameters;
+import core.components.Component;
+import core.components.PartialObservableDeck;
+import core.interfaces.IGamePhase;
+import games.GameType;
+import games.resistance.actions.ResMissionVoting;
+import games.resistance.actions.ResTeamBuilding;
+import games.resistance.actions.ResVoting;
+import games.resistance.components.ResGameBoard;
+import games.resistance.components.ResPlayerCards;
+import org.apache.avro.generic.GenericData;
+
+import java.util.*;
+
+public class ResGameState extends AbstractGameState {
+
+    public int[] factions;
+    List<Boolean> gameBoardValues = new ArrayList<>();
+    List<Integer> noVotesPerMission = new ArrayList<>();
+    boolean voteSuccess;
+    int leaderID;
+    int failedVoteCounter = 0;
+    Random rnd;
+
+    ResPlayerCards.CardType[] votingChoice;
+    ResPlayerCards.CardType[] missionVotingChoice;
+
+    List<Integer> teamChoice;
+    IGamePhase previousGamePhase = null;
+    List<Integer> finalTeamChoice = new ArrayList<>();
+    List<List<Integer>> historicTeams = new ArrayList<>();
+
+    public enum ResGamePhase implements IGamePhase {
+        MissionVote, TeamSelectionVote, LeaderSelectsTeam
+    }
+
+    List<PartialObservableDeck<ResPlayerCards>> playerHandCards = new ArrayList<>(10);
```

Comment:
I think this is better separated, with 1 ID/alliance flag for each player, and one voting array. I'm not sure if it needs to go through a PO deck with 3 cards per player with specific access at idx 2 to get a player's identity for example, it feels overcomplicated. Isn't the voting in a separate array already anyway? 

Then the card extending class could just be an enum, since it doesn't really have any other role?

Reply:
I agree it's sub-optimal (the use of Cards), but I don;t want to change it as it's also deeply entwined in the Resistance GUI. (And it works.)

However, voting now refactored as you suggest to a single mechanism.